api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsServiceImpl.java

```diff
@@ -203,6 +197,110 @@ public ResponseEntity updateTheRemissionAmount(String paymentReference, Resubmit
         return new ResponseEntity<>(null, HttpStatus.OK);
     }
 
+    @Override
+    public PaymentGroupResponse checkRefundAgainstRemission(MultiValueMap<String, String> headers,
+                                                            PaymentGroupResponse paymentGroupResponse, String ccdCaseNumber) {
+
```

Comment:
Some log statements would be very helpful along with comments as to what each block of code is doing.

Reply:
Added comments and logs